src/bin/cargo/cli.rs

```diff
@@ -571,6 +571,8 @@ pub fn cli(gctx: &GlobalContext) -> Command {
             .invalid(style::INVALID)
     };
 
+    let alias_map = super::user_defined_aliases(gctx);
```

Comment:
This is pessimizing performance.  We'll load user defined aliases an extra time for no other reason than for when completions are performed.  This should be called inside of the closure.  If you look at the ArgValueCompleters in `command_prelude.rs`, you can see how we deal with getting a global context inside of a completer